metadeploy/api/models.py

```diff
@@ -117,6 +120,9 @@ def is_visible_to(self, user):
             )
         )
 
+    def is_visible_to_scratch_org(self):
+        return not self.visible_to or ORG_TYPES.Scratch in self.visible_to.org_type
+
```

Comment:
@davisagli It seems like we might want to make some adjustments to the `AllowedListAccessMixin` so that anonymous users can access products/plans that are restricted to `org_type=["Scratch"]`? I'm not 100% clear on the use-case there, so I broke it out into a new story to consider that. https://trello.com/c/V8FwiNnF